### PR TITLE
 bug fix: modify assert->BOOST_THROW_EXCEPTION when the assert condition is decided by outside-users and maybe cause the program coredump

### PR DESCRIPTION
**Pull request by cyjseagull**

### CODE REVIEW

File: libevm/EVMC.cpp [hunk 1]
```diff
@@ -45,15 +46,32 @@ EVM::EVM(evmc_instance* _instance) noexcept : m_instance(_instance)
...
+    /// gas, gasLimit are u256 outside of EVM
+    /// and can be passed inside by the outsider-users
+    /// so we should remove the `assert`, and throw exceptions if this happend
+    /// *origin code:
+    /// assert(io_gas <= int64max);
+    /// assert(_ext.envInfo().gasLimit() <= int64max);
+    /// *modified code:
+    if (io_gas > int64max || _ext.envInfo().gasLimit() > int64max)
+    {
+        LOG(ERROR) << LOG_DESC("Gas overflow") << LOG_KV("cur gas", io_gas)
```
Comment (bxq2011hust) [inline libevm/EVMC.cpp]: cur gas -> current gas

File: libevm/EVMC.cpp [hunk 2]
```diff
@@ -45,15 +46,32 @@ EVM::EVM(evmc_instance* _instance) noexcept : m_instance(_instance)
...
+                   << LOG_KV("cur gasLimit", _ext.envInfo().gasLimit())
```
Comment (bxq2011hust) [inline libevm/EVMC.cpp]: cur gasLimit -> gasLimit

File: libevm/EVMC.cpp [hunk 3]
```diff
@@ -45,15 +46,32 @@ EVM::EVM(evmc_instance* _instance) noexcept : m_instance(_instance)
...
+                   << LOG_KV("max gas/gasLimit", int64max);
```
Comment (bxq2011hust) [inline libevm/EVMC.cpp]: don't use /?

File: libevm/EVMC.cpp (hunk 1 above)
Comment (cyjseagull) [reply libevm/EVMC.cpp]: done.

File: libevm/EVMC.cpp (hunk 3 above)
Comment (cyjseagull) [reply libevm/EVMC.cpp]: done.

Comment (cyjseagull) [reply libevm/EVMC.cpp]: done.

**Pull request merged**